datafusion/common/src/scalar.rs

```diff
@@ -637,14 +643,8 @@ impl std::hash::Hash for ScalarValue {
                 s.hash(state)
             }
             Boolean(v) => v.hash(state),
-            Float32(v) => {
-                let v = v.map(OrderedFloat);
-                v.hash(state)
-            }
-            Float64(v) => {
-                let v = v.map(OrderedFloat);
-                v.hash(state)
-            }
+            Float32(v) => v.map(Fl).hash(state),
```

Comment:
I think this can just call HashValue on v?

Reply:
Fair, I think there is a way to clean this up, but we can do that in a follow on PR